app/core/services/ProcessPopulation.scala

```diff
@@ -22,6 +22,7 @@ import core.models.ocelot.errors._
 import scala.annotation.tailrec
 
 trait ProcessPopulation {
+  this: PlaceholderProvider =>
```

Comment:
What does this do?

Reply:
means "this" has to be a PlaceholderProvider i.e. ProcessPopulation can only be mixed into a class which implements PlaceholderProvider

Reply:
Thanks you. Also helps explain why PageBuilder needs an instance of PlaceHolders.